ci-operator/step-registry/cucushift/installer/check/aws/private/cucushift-installer-check-aws-private-commands.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+REGION="${LEASED_RESOURCE}"
+INFRA_ID=$(jq -r '.infraID' ${SHARED_DIR}/metadata.json)
+CLUSTER_NAME="${NAMESPACE}-${UNIQUE_HASH}"
+
+export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
+
+ret=0
+
+#Check there's no external load balancer created in this cluster
+CLUSTER_LB_LIST=$(mktemp)
+
+for lb in $(aws --region ${REGION} elbv2 describe-load-balancers --no-paginate | jq -r '.LoadBalancers[].LoadBalancerArn'); do
+    aws --region ${REGION} elbv2 describe-tags --resource-arns "$lb" | jq --arg infra "kubernetes.io/cluster/${INFRA_ID}" -ce '.TagDescriptions[].Tags[] | select( .Key == "\($infra)")' && echo "$lb" >> "${CLUSTER_LB_LIST}"
+done
+
+for cluster_lb in $(cat $CLUSTER_LB_LIST); do 
+    LB_TYPE=$(aws --region ${REGION} elbv2 describe-load-balancers --load-balancer-arns ${cluster_lb} | jq -r '.LoadBalancers[].Scheme')
+    if [[ "${LB_TYPE}" == "internal" ]]; then
+        echo "Pass: LB ${cluster_lb} is internal."
+    else
+        echo "Error: LB ${cluster_lb} is not internal, the LB type is ${LB_TYPE}"
+        ret=$((ret + 1))
+    fi
+done
+
+
+#No public Route 53 DNS records that matches the baseDomain for the cluster
+if [[ -z ${BASE_DOMAIN} ]]; then
+  echo "Error: BASE_DOMAIN is not set, exit."
+  exit 1
+fi
+
+PUBLIC_ZONE_ID=$(aws route53 list-hosted-zones-by-name | jq --arg name "${BASE_DOMAIN}." -r '.HostedZones | .[] | select(.Name=="\($name)") | .Id' | awk -F / '{printf $3}')
+
+if [[ -n "${PUBLIC_ZONE_ID}" ]]; then
+    aws route53 list-resource-record-sets --hosted-zone-id=${PUBLIC_ZONE_ID} --output json | jq '.ResourceRecordSets[].Name' |grep "${CLUSTER_NAME}.${BASE_DOMAIN}" && ret=$(( $ret | 2 ))
+
+    if [ ${ret} -ge 2 ]; then
+        echo "Error: there's public Route 53 DNS records that matches this cluster"
+    else
+        echo "PASS: no public Route 53 DNS records that matches this cluster"
+    fi
+else
+    echo "Error: no valid PUBLIC_ZONE_ID found for this base domain ${BASE_DOMAIN}, exit."
```

Comment:
I think here we can skip the check, because the BASE_DOMAIN maybe some random string, there is no existing public zone. It does not affect private cluster install.

Reply:
ok, that makes sense